File: problem2.js
#!/usr/local/bin/node

function even_fib(p1, p2, cap) {
  var c = p1 + p2;
  var a = c % 2 == 0 ? c : c == 3 ? p2 : 0;
  if (c < cap ) {
   return a + even_fib(p2, c, cap);
  }
  return a;
}

var sum = even_fib(1,2,4000000);
console.log(sum);

// 4613732

File: problem2.clj
(defn even_fib [p1, p2 cap]
  (def c (+ p1 p2))
  (def a
    (if (= 0 (mod c 2)) c
        (if (= 3 c) p2 0)))
  (if (<  (+ p1 p2) cap) (+ a (even_fib p2 c cap))
      a))
(println (even_fib 1 2 4000000))
